### PR TITLE
[EISW-44218] modify cmake/developer_package/version.cmake add macro, …

### PR DESCRIPTION
**Pull request by weqaar**
### Details:
 - *cmake/dependencies.cmake: WIN32 OpenCV version and associated SHA to 4.5.5*
 - *cmake/developer_package/version.cmake: AIRA [EISW-41342]*
 
### Tickets:
 - *EISW-44218*
 - *EISW-41342*
 - *EISW-43796*
 - *EISW-43363*
 - *EISW-41665*

### CODE REVIEW

File: cmake/developer_package/version.cmake [hunk 1]
```diff
@@ -94,20 +94,27 @@ macro(ie_parse_ci_build_number)
...
+        commitHash(GIT_COMMIT_HASH)
+        if(${IE_VERSION} EQUAL 0)
+            set(CI_BUILD_NUMBER "custom_${GIT_BRANCH}_${GIT_COMMIT_HASH}")
+        else()
+            set(CI_BUILD_NUMBER "${IE_VERSION}_${GIT_BRANCH}_${GIT_COMMIT_HASH}")
+        endif()
+    endif()
+endmacro()
+
+ie_set_version(0)
```
Comment (ilyachur) [inline cmake/developer_package/version.cmake:111]: Why do you need two calls?

File: cmake/developer_package/version.cmake [hunk 2]
```diff
@@ -94,20 +94,27 @@ macro(ie_parse_ci_build_number)
...
+#[[ Provides Inference Engine version via either of the two:
+    1. Use CI_BUILD_NUMBER if defined in env
+    2. Parse ie_version.hpp
+]]
```
Comment (ilyachur) [inline cmake/developer_package/version.cmake:115]: ??

```suggestion
#[[ Provides Inference Engine version via either of the two:
#    1. Use CI_BUILD_NUMBER if defined in env
#    2. Parse ie_version.hpp
#]]
```

File: cmake/developer_package/version.cmake [hunk 3]
```diff
@@ -94,20 +94,27 @@ macro(ie_parse_ci_build_number)
...
-    commitHash(GIT_COMMIT_HASH)
-
-    set(custom_build "custom_${GIT_BRANCH}_${GIT_COMMIT_HASH}")
-    set(CI_BUILD_NUMBER "${custom_build}")
-endif()
-
-# provides Inference Engine version
-# 1. If CI_BUILD_NUMBER is defined, parses this information
-# 2. Otherwise, parses ie_version.hpp
+macro (ie_set_version IE_VERSION)
```
Comment (ilyachur) [inline cmake/developer_package/version.cmake:97]: Why do we need this change?

**Comment by ilyachur** [comment]
@ilya-lavrenov please take a look

File: cmake/developer_package/version.cmake (hunk 1 above)
Comment (weqaar) [reply cmake/developer_package/version.cmake:111]: First call sets CI_BUILD_NUMBER to to custom_ format required by macro(ie_parse_ci_build_number):
if(${IE_VERSION} EQUAL 0)
            set(CI_BUILD_NUMBER "custom_${GIT_BRANCH}_${GIT_COMMIT_HASH}")
Second call, sets the correct version string with: 
            set(CI_BUILD_NUMBER "${IE_VERSION}_${GIT_BRANCH}_${GIT_COMMIT_HASH}")
This is required for applications like AIRA to function correctly as they require OpenVINO version as the first part of string in format above, see EISW-41342 for details.

File: cmake/developer_package/version.cmake (hunk 3 above)
Comment (weqaar) [reply cmake/developer_package/version.cmake:97]: This is required for applications like AIRA to function correctly as they require OpenVINO version as the first part of string in format above, see EISW-41342 for details.

File: cmake/developer_package/version.cmake (hunk 1 above)
Comment (ilya-lavrenov) [reply cmake/developer_package/version.cmake:111]: @weqaar does this https://github.com/openvinotoolkit/openvino/pull/12521 work for you?

File: cmake/dependencies.cmake [hunk 4]
```diff
@@ -220,8 +220,8 @@ endfunction()
 if(ENABLE_OPENCV)
     reset_deps_cache(OpenCV_DIR)
 
-    set(OPENCV_VERSION "4.5.2")
-    set(OPENCV_BUILD "076")
+    set(OPENCV_VERSION "4.5.5")
+    set(OPENCV_BUILD "099")
```
Comment (ilya-lavrenov) [inline cmake/dependencies.cmake:224]: why do we need to change OpenCV version?

File: cmake/developer_package/version.cmake (hunk 1 above)
Comment (weqaar) [reply cmake/developer_package/version.cmake:111]: yes it should, we have test AIRA with it, will confirm

**Pull request closed without merging**